quickwit/quickwit-ingest/src/ingest_v2/gc.rs

```diff
@@ -0,0 +1,225 @@
+// Copyright (C) 2023 Quickwit, Inc.
+//
+// Quickwit is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at hello@quickwit.io.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use std::collections::HashMap;
+use std::sync::Arc;
+use std::time::Duration;
+
+use mrecordlog::error::DeleteQueueError;
+use mrecordlog::MultiRecordLog;
+use once_cell::sync::Lazy;
+use quickwit_proto::ingest::IngestV2Result;
+use quickwit_proto::metastore::{DeleteShardsRequest, DeleteShardsSubrequest};
+use quickwit_proto::{split_queue_id, IndexUid, QueueId, ShardId, SourceId};
+use tokio::sync::{RwLock, Semaphore};
+use tracing::{error, info};
+
+use super::ingest_metastore::IngestMetastore;
+use super::ingester::IngesterState;
+
+/// Period of time after which shards candidate for deletion are actually deleted.
```

Comment:
It would be nicer to explicit what we mean by deleting shards here:
Of course this is not about truncating data. At this point hopefully the data has been deleted, but this is about deleting from the metastore.

Reply:
Also I would but the last two lines of description on the GcTask doc instead of. the GRACE_PERIOD doc